lib/index.js

```diff
@@ -206,24 +206,41 @@ api.inboxes.getAll = function(actor, query, fields, options, callback) {
 
   query = query || {};
   fields = fields || {};
+  // meta is required for permission check, if meta was
+  let stripMeta = false;
+  if(Object.keys(fields).length > 0 && !fields.meta) {
+    stripMeta = true;
+    fields.meta = true;
+  }
   options = options || {};
   async.auto({
-    // need inbox access capability for all resources to use this
-    checkPermission: callback => brPermission.checkPermission(
-      actor, PERMISSIONS.LDN_INBOX_ACCESS, callback),
-    find: ['checkPermission', (results, callback) => inboxCollection.find(
-      query, fields, options).toArray(callback)]
+    find: callback => inboxCollection.find(query, fields, options)
+      .toArray(callback),
+    // check to make sure the caller is allowed to access the documentType
+    // (Note: fields *must not* have excluded `documentType.id` in this
+    // case or else the look up for the permission check will fail)
+    getAuthorized: ['find', (results, callback) => async.filterSeries(
+      results.find, (record, callback) => brPermission.checkPermission(
+        actor, PERMISSIONS.LDN_INBOX_ACCESS, {
+          resource: record.meta,
```

Comment:
The resource isn't `record.meta` ... I find this confusing. It should be `record.inbox.id`. If we've already pulled out `meta`, we can just use a list: `resource: [record.inbox.id, record.meta.owner]` and don't use `translate`.